lib/ansible/modules/crypto/openssl_pkcs12.py

```diff
@@ -363,10 +415,13 @@ def main():
                 if module.params['action'] == 'export':
                     if not module.params['friendly_name']:
                         module.fail_json(msg='Friendly_name is required')
-                    pkcs12.generate(module)
+                    pkcs12_content = pkcs12.generate(module)
+                    pkcs12.write(module, pkcs12_content, 0o600)
                     changed = True
                 else:
-                    pkcs12.parse(module)
+                    pkey, cert, ca_certs, friendly_name = pkcs12.parse()
+                    dump_content = '%s%s%s' % (to_native(pkey), to_native(cert), b''.join(ca_certs))
```

Comment:
```suggestion
                    dump_content = '%s%s%s' % (to_native(pkey), to_native(cert), to_native(b''.join(ca_certs)))
```
Otherwise the last line of the PEM file will be `b'-----BEGIN CERTIFICATE-----\nMIIE4zCCAsugAwIBAgIUerXv6Q7bEPOTRmAehOZkVavpUz0wDQYJKoZIhvcNAQEL\nBQAwGzEZMBcGA1UEAwwQd3d3...` (at least in Python 3).